c7n/resources/s3.py

```diff
@@ -371,6 +371,65 @@ def process_bucket(self, b):
             return b
 
 
+@filters.register('access')
+class AccessFilter(Filter):
+    """Filters for all S3 buckets that have access allowed/denied to a role
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: s3-perm-check
+                resource: s3
+                filters:
+                  - type: access
+                    value: denied
+                    policy_source: arn:::
+    """
+
+    schema = type_schema('access',
+        policy_source={'type': 'string'},
+        caller={'type': 'string', 'required': False},
+        actions={'type': 'array', 'items': {'type': 'string'}},
+        value={'type': 'string', 'enum': ['allowed', 'denied']})
+
+    permissions = ('s3:ListBucket',)
```

Comment:
Done in d4e29d4e64170a5dd46f2450057e38621430d58e. We're only simulating principal policies at this point, not custom.